src/patterns/scheduled-ecs-task.ts

```diff
@@ -0,0 +1,239 @@
+import { Alarm, TreatMissingData } from "@aws-cdk/aws-cloudwatch";
+import { SnsAction } from "@aws-cdk/aws-cloudwatch-actions";
+import type { ISecurityGroup, IVpc } from "@aws-cdk/aws-ec2";
+import type { IRepository } from "@aws-cdk/aws-ecr";
+import {
+  Cluster,
+  Compatibility,
+  ContainerImage,
+  FargatePlatformVersion,
+  LogDrivers,
+  TaskDefinition,
+} from "@aws-cdk/aws-ecs";
+import type { Schedule } from "@aws-cdk/aws-events";
+import { Rule } from "@aws-cdk/aws-events";
+import { SfnStateMachine } from "@aws-cdk/aws-events-targets";
+import { PolicyStatement } from "@aws-cdk/aws-iam";
+import { Topic } from "@aws-cdk/aws-sns";
+import { IntegrationPattern, StateMachine } from "@aws-cdk/aws-stepfunctions";
+import { EcsFargateLaunchTarget, EcsRunTask } from "@aws-cdk/aws-stepfunctions-tasks";
+import { CfnOutput, Duration } from "@aws-cdk/core";
+import { GuDistributionBucketParameter } from "../constructs/core";
+import type { GuStack } from "../constructs/core";
+import type { Identity } from "../constructs/core/identity";
+
+/**
+ * Configuration to determine what container to use for the task
+ * For example, to run the task in the latest node container:
+ *
+ * ```typescript
+ * const containerConfiguration = {
+ *     id: 'node:latest'
+ * }
+ * ```
+ *
+ * Alternatively you can specify a repository and version, for example for an ECR repository:
+ * ```typescript
+ * import { Repository } from "@aws-cdk/aws-ecr";
+ * const repository = new Repository(scope, `${app}-repository`, {
+    repositoryName: app,
+  });
+ * const containerConfiguration = {
+ *     repository: Repository.fromRepositoryArn("<repository arn>"),
+ *     version: '1'
+ * }
+ * ```
+ */
+interface ContainerConfiguration {
+  id?: string;
+  repository?: IRepository;
+  version?: string;
+}
```

Comment:
Should all these properties be optional? Would it be clearer to make the usage of `ContainerConfiguration` be of type `ContainerConfiguration | undefined`?

Reply:
Good point. I think I've made things clearer in https://github.com/guardian/cdk/pull/791/commits/cb6a6fb20db19655d6410080a996c10b06517856

Reply:
Much nicer! 👍🏽